sky/shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -472,13 +473,50 @@ public void runFromBundle(String bundlePath, String snapshotPath) {
         postRun();
     }
 
-    public void runFromFile(String main, String packageRoot) {
-        preRun();
-        mSkyEngine.runFromFile(main, packageRoot, "");
-        postRun();
+    private class RunFromSourceRunnable implements Runnable {
+        RunFromSourceRunnable(String main,
+                              String packages,
+                              String assetsDirectory,
+                              FlutterView flutterView) {
+            this.main = main;
+            this.packages = packages;
+            this.assetsDirectory = assetsDirectory;
+            this.flutterView = flutterView;
+        }
+
+        private String main;
+        private String packages;
+        private String assetsDirectory;
+        private FlutterView flutterView;
```

Comment:
The RunFromSourceRunnable inner class will have an implicit reference to the outer FlutterView.

You can make the arguments to runFromSource final and reduce RunFromSourceRunnable to an anonymous inner class within runFromSource.